### PR TITLE
Make azdo token optional for local scenario test run

### PR DESCRIPTION
**Pull request by oleksandr-didyk**
Resolves: https://github.com/dotnet/arcade-services/issues/4012

Build: https://dev.azure.com/dnceng/internal/_build/results?buildId=2550068&view=results

### CODE REVIEW

File: test/Maestro.ScenarioTests/TestParameters.cs [hunk 1]
```diff
@@ -177,12 +177,17 @@ private TestParameters(
 
     public string AzureDevOpsProject { get; } = "internal";
 
-    public string AzDoToken { get; }
+    public AzureDevOpsTokenProvider AzdoTokenProvider { get; }
```
Comment (dkurepa) [inline test/Maestro.ScenarioTests/TestParameters.cs:180]: can this be private, since we have the method below?

Comment (oleksandr-didyk) [reply test/Maestro.ScenarioTests/TestParameters.cs:180]: Yeah, it can be, will change

**Review by dkurepa — APPROVED** [review]

**Comment by dkurepa** [comment]
forgot about this in the PR, but can we also port this to the PCS scenario tests?

**Comment by oleksandr-didyk** [comment]
> forgot about this in the PR, but can we also port this to the PCS scenario tests?

Of course, I'll get to it

**Comment by oleksandr-didyk** [comment]
> > forgot about this in the PR, but can we also port this to the PCS scenario tests?
> 
> Of course, I'll get to it

PCS scenario tests already have the AzDO token as non-required, so it was just Maestro scenario tests that were left

**Pull request merged**